lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,301 @@
+    """
+    @jwt_required
+    def post(self):
+        """
+        Post request for logging out a user.
+        This requires a JWT auth token in the header field
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+            X-Auth-Token:   JWT token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"/"fail",
+                    "status_code": 200/status_code_int,
+                    "message": "Successfully logged out."/"failure message",
+                }
+        """
+        # get auth token
+        auth_header = request.headers.get("X-Auth-Token")
+        if auth_header:
+            auth_token = auth_header.split(" ")[1]
+
+            user_id = UserModel.decode_auth_token(auth_token)
+            if not isinstance(user_id, str):
+                # Add the token to the blacklist
+                blacklist.add(auth_token)
+                response_object = {
+                    "status": "success",
+                    "message": "Successfully logged out.",
+                    "status_code": 200,
+                }
+                return make_response(jsonify(response_object))
+            else:
+                response_object = {
+                    "status": "fail",
+                    "message": user_id,
+                    "status_code": 401,
+                }
+                return make_response(jsonify(response_object))
+        else:
+            response_object = {
+                "status": "fail",
+                "message": "Provide a valid auth token.",
+                "status_code": 403,
+            }
+            return make_response(jsonify(response_object))
+
+
+class GetUser(Resource):
+    """
+    Abstracted pbench API to get user metadata information
+    """
+
+    # TODO: We can implement the graphql query to get specific metadata related to a user
+    # We dont need to pass user_id to get the user, user id will be retrieved from the jwt encoded auth header
+    @jwt_required
+    def get(self):
+        """
+        Get request for getting user metadata.
+        This requires a JWT auth token in the header field
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+            X-Auth-Token:   JWT token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"/"fail",
+                    "status_code": 200/status_code_int,
+                    "message": "Successfully registered."/"failure message",
+                    "data": {
+                        "username": username,
+                        "registered_on": registered_on,
+                    }
+                }
+        """
+        # get the auth token
+        auth_header = request.headers.get("X-Auth-Token")
```

Comment:
I changed this in my API, after finding out that JWT recommends using the `Authorization` header with the `Bearer` schema: e.g., `Authorization: Bearer <token>`. (Case insensitive, of course.) It's a bit more work to do the split to get the token, but to the extent there's a standard (and it's a loose recommendation in this case), I'd rather follow it. Actually, I see you're already doing the split below, but not checking for a schema; which in any case would be wildly unstandardized for `X-Auth-Token` which is usually just a single token.

There should be a separate method to extract the JWT fields; probably even something that returns the JWT token from a `request.headers` dict with proper error checking. Then we can use it elsewhere throughout the API.

Reply:
I see thanks for the clarification, we would still have to do something like `X-Auth-Token Bearer <token>` so thats why the split, but I will update the auth header as `Authorization`